hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -69,7 +77,6 @@ public class HoodieTableFactory implements DynamicTableSourceFactory, DynamicTab
   public DynamicTableSource createDynamicTableSource(Context context) {
     Configuration conf = FlinkOptions.fromMap(context.getCatalogTable().getOptions());
     ResolvedSchema schema = context.getCatalogTable().getResolvedSchema();
-    sanityCheck(conf, schema);
     setupConfOptions(conf, context.getObjectIdentifier(), context.getCatalogTable(), schema);
```

Comment:
We can not remove the sanity check for MOR table source, the MOR table relies on the pk and preCombine key for merging payloads.

Reply:
Oh i miss that pk field is used to emit delete data.
I add sanity check for stream read mor table.

Reply:
Yeah, mering happens for mor table no matter whether the consuming is streaming or batch.

Reply:
But as in the code of MergeOnReadInputFormat.MergeIterator.hasNext(), record key is read from meta field, and in constructor of AbstractHoodieLogRecordReader the preCombineField is read from table config.
So it seems that the passed in recordKey is only used to emit delete record when stream read mor log and preCombine field is not used at all when reading hudi table?

Reply:
You are right